### ISSUE TITLE
Rez env strips double dash in command

### ISSUE THREAD

**Issue by johhnry** [AcademySoftwareFoundation/rez#1462]
Hi,

I want to pass arguments to a Blender Python script through the command line as described [here](https://blender.stackexchange.com/questions/6817/how-to-pass-command-line-arguments-to-a-blender-python-script).

I should be able to do this using Rez:

```shell
$ rez env blender -- blender -b --python script.py -- -s "test"
```

It works when running it in a new Rez shell but not when running the command directly with double dash.

**Environment**
* Windows 10
* Rez 2.111.2
* Rez Python 3.7.7

**To Reproduce**

```python
# test.py
import sys

if __name__ == "__main__":
    print(sys.argv)
```

Running the command:

```shell
$ rez env python -- python test.py -- -s "test"
```

**Expected behavior**

Should print the following:

```
['test.py', '-s', 'test']
```

**Actual behavior**

Instead it strips out the arguments after the `--` (double dashes):

```
['test.py']
```

**Comment by JeanChristopheMorinPerso** [comment]
Hi @johhnry, I think you can just do `rez env python -- python test.py -s "test"` and it'll work. When I test locally it works as expected:

```
[jcmorin@arch01 /tmp]$ rez-env python -- python test.py -s "asd"
['test.py', '-s', 'asd']
```

Or do you have a specific reason to need the second `--`?

**Comment by johhnry** [comment]
Hi  @JeanChristopheMorinPerso,

The specific reason to use `--` is passing arguments to a Blender python script like I described earlier:

```
$ rez env blender -- blender -b --python script.py -- -s "test"
```

Since if I pass them after the `--python script.py`, they will be parsed as Blender arguments. (Blender ignore all arguments after the `--` so it's convenient to pass them here)

**Comment by instinct-vfx** [comment]
What platform / shell are you using? If i am not mistaken `--` is a shell feature. Can you maybe escape it?

**Comment by JeanChristopheMorinPerso** [comment]
I know that we do control `--` in Rez. Not sure if it works I cmd though, but I am under the impression it works in PowerShell...

**Comment by johhnry** [comment]
@instinct-vfx 
I am using Powershell in Windows Terminal but it also does the same behavior in cmd:

```shell
# In a Rez resolved shell (cmd)
$ blender -b --python test.py -- -s "test"
['blender', '-b', '--python', 'test.py', '--', '-s', 'test']

# In cmd
$ rez env blender -- blender -b --python test.py -- -s "test"
['blender', '-b', '--python', 'test.py']
```

**Comment by johhnry** [comment]
@instinct-vfx 
I am using Powershell in Windows Terminal but it also does the same behavior in cmd:

```shell
# In a Rez resolved shell (cmd)
$ blender -b --python test.py -- -s "test"
['blender', '-b', '--python', 'test.py', '--', '-s', 'test']

# In cmd
$ rez env blender -- blender -b --python test.py -- -s "test"
['blender', '-b', '--python', 'test.py']
```

**Comment by johhnry** [comment]
I am using Powershell in Windows Terminal but it also does the same behavior in cmd:

```shell
# In a Rez resolved shell (cmd)
$ blender -b --python test.py -- -s "test"
['blender', '-b', '--python', 'test.py', '--', '-s', 'test']

# In cmd
$ rez env blender -- blender -b --python test.py -- -s "test"
['blender', '-b', '--python', 'test.py']
```

**Comment by JeanChristopheMorinPerso** [comment]
Yes, I expect the behavior to be the same. `--` is handled by the program, not the OS.

**Comment by dbr** [comment]
I ran into this with nested `rez env` commands, for example

```sh
$ rez env python -- rez env python -- python3
```

..which I would expect to execute the `python3` command, however it just runs the second `rez env` command without arguments(?)

As I understand, it should function the same as either of these two options:
```sh
$ rez env python
> $ rez env python
>> $ python 3
```
or
```sh
$ rez env python -c 'rez env python -c "python"'
```

**Comment by herronelou** [comment]
I have run into the same issue and had posted about it on Slack, adding the link here for posterity.

https://academysoftwarefdn.slack.com/archives/C0321B828FM/p1694028568579839

**Comment by dbr** [comment]
Ran into this issue again today, using the Rez integration in Deadline.

I tried to submit a command line job which ran something like `rez env example_pkg -- example`

However, the included-by-default Rez event plugin for Deadline essentially does

    rez env $packages_from_submission etc -- $user_command

So, with `$user_command` is the above `example` command, what will be executed is:

    rez env $packages_from_submission etc -- rez env example_pkg -- example

which of course triggers this bug (resulting in the task executing an interactive shell and getting stuck forever)

**Comment by Anti-Distinctlyminty** [comment]
Had a recent discussion over on Blender issues [Blender issues](https://projects.blender.org/blender/blender/issues/126818) which turned out to be this bug. In short, when trying to use the graphics cards to render via command line one must execute a command line similar to

```
blender.exe .\my.blend -x 1 --background -E CYCLES --render-anim --frame-start 1 --frame-end 2 -- --cycles-device OPTIX
```

What ends up executing when in a Rez context

```
blender.exe .\my.blend -x 1 --background -E CYCLES --render-anim --frame-start 1 --frame-end 2
```

**Comment by instinct-vfx** [comment]
That's not really a bug though, is it. `--` is used to indicate the end of options. See also https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_02

> Guideline 10: The first -- argument that is not an option-argument should be accepted as a delimiter indicating the end of options. Any following arguments should be treated as operands, even if they begin with the '-' character.

Can you maybe escape it or use -c instead of `--`?

I guess this was done "by design" though it likely does not really propagate well to non-linux platforms.

**Comment by herronelou** [comment]
There is a definite bug as rez just swallows them in certain situations. I
thought I had a PR for this but maybe I never put it through it's been a
while.

**Comment by vanridal** [comment]
I think is a valid issue, probably not a bug but a lack of implementation, this is blocker for me when wanting to use nested `--` without any arg encapsulation.

On further investigation, rez does handle another exists of a `--` for a reason
**rez-build**
```
  --ba ARGS, --build-args ARGS
                        arguments to pass to the build system. Alternatively, list these after a '--'.
  --cba ARGS, --child-build-args ARGS
                        arguments to pass to the child build system, if any. Alternatively, list these after a second '--'
```
But i feel that doesnt mean `rez-env` has to parse it this way
currently all existences of `--` split the args into a list of lists called `extra_arg_groups`
and `rez-env` just takes the first index
`command = extra_arg_groups[0] or None`

Each command plugin can handle `extra_arg_group` in its own way, the question I pose is does rez want to allow `rez-env` to flatten all arg groups so that any further arg separation is handled by the rez environment sub-process, what benefit does `rez-env` have not to pass the other arg groups through?
